static/js/components/CommentList.jsx

```diff
@@ -377,6 +403,28 @@ const CommentList = ({
                             <CloseIcon fontSize="small" />
                           </Button>
                         )}
+                        {associatedResourceType === "shift" && (
+                          <Button
+                            style={
+                              hoverID === id
+                                ? {
+                                    display: "block",
+                                    minWidth: "0",
+                                    lineHeight: "0",
+                                    padding: "0",
+                                  }
+                                : { display: "none" }
+                            }
+                            size="small"
+                            color="primary"
+                            type="button"
+                            name={`deleteCommentButtonShift${id}`}
+                            onClick={() => deleteCommentOnShift(id)}
+                            className="commentDelete"
+                          >
+                            <CloseIcon fontSize="small" />
+                          </Button>
+                        )}
```

Comment:
since this code is repeated twice, would it make sense to extract into its own component? not exactly related to this PR, but I find the component hard to read since it's wrapped in a big if based on if the comments are compact or not. I think it would be better to split into two components, compact comment list and regular comment list, and extract out some of the components that would be common to both like the delete button. 

Also, it would be nice if there was less duplicated code for checking the associatedResourceType. I think some of that could be extracted out into helper functions which check the associatedResourceType. So instead of having four delete button declarations, there would just be one. Some thoughts about this component as a whole, but if you don't want to change it in this PR we should make an issue to clean this up.

Reply:
Yeah, it would be nicer to have one component, to which maybe you can pass the name and the onClick function. Maybe even the className.